statsmodels/tsa/base/tsa_model.py

```diff
@@ -220,7 +227,8 @@ def _init_dates(self, dates=None, freq=None):
         # Construct the internal index
         index_generated = False
 
-        if (date_index and has_freq) or (int_index and is_increment):
+        if ((date_index and has_freq) or (int_index and is_increment) or
+                range_index):
```

Comment:
Should have asked about the index.copy() here.

Also, w/r/t 236-241, are all of these attributes necessary?  ("Out Of Scope/Not My Problem is an acceptable answer here")  `_index_inferred_freq` is never used.  `_index_generated` is used, but it isn't entirely obvious why (also it is used identically in three places, needs to be made a function)

Reply:
I think these attributes are useful.

> _index_generated is used, but it isn't entirely obvious why

It is necessary to know if the user provided an index that cannot be used for forecasting (in which case the forecasting index is a generated index). This is because in-sample prediction can still be done using keys from the provided index.

> (also it is used identically in three places, needs to be made a function)

I don't understand this comment.

Reply:
> I think these attributes are useful.

Does that include _index_inferred_freq?  Because that's never used.  The what-attributes-exist guessing game is one of my pet peeves in sm, so I'm pushing to trim unused attributes where possible.


>> (also it is used identically in three places, needs to be made a function)

> I don't understand this comment.

sarimax:
            # Reset the ModelData datasets and cache
            self.data.endog, self.data.exog = (
                self.data._convert_endog_exog(endog, exog))

            # Reset indexes, if provided
            new_length = self.data.endog.shape[0]
            if self.data.row_labels is not None:
                self.data._cache['row_labels'] = (
                    self.data.row_labels[orig_length - new_length:])
            if self._index is not None:
                if self._index_generated:
                    self._index = self._index[:-(orig_length - new_length)]
                else:
                    self._index = self._index[orig_length - new_length:]
```

markov_autoregression
```
        # Reset indexes, if provided
        if self.data.row_labels is not None:
            self.data._cache['row_labels'] = (
                self.data.row_labels[self.order:])
        if self._index is not None:
            if self._index_generated:
                self._index = self._index[:-self.order]
            else:
                self._index = self._index[self.order:]
```

arima_model
```
        # Reset index
        orig_length = len(self._index)
        new_length = self.endog.shape[0]
        if self.data.row_labels is not None:
            self.data._cache['row_labels'] = (
                self.data.row_labels[orig_length - new_length:])
        if self._index is not None:
            if self._index_generated:
                self._index = self._index[:-(orig_length - new_length)]
            else:
                self._index = self._index[orig_length - new_length:]
```

Reply:
> Does that include _index_inferred_freq? Because that's never used.

Good question, but it's the only way to tell if the index has an inferred frequency versus one explicitly passed by the user, and also has a self-explanatory name, so I think it's okay to keep.